### PR TITLE
Fix ndef as code2

### PR DESCRIPTION
**Pull request by adcxyz**
## Purpose and Motivation

This is a cleanup PR that replaces #4469, which was cluttered with accediental commits. 
Fixes the bug described in #4329, where nodeproxy/Ndef posting as code fails in some cases.
Please read full background in #4469 

## Types of changes

- Bugs fixed
- Tests added for all cases
- Documentation updated

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review

### CODE REVIEW

File: HelpSource/Tutorials/JITLib/jitlib_asCompileString.schelp [hunk 1]
```diff
@@ -47,3 +47,43 @@ p.document;		// document everything
...
+Ndef(\a,  { Saw.ar(\freq.kr, 0.1) }).set(\freq, 123);
+Ndef(\a).asCode;
+// returns:
+"(
+Ndef('a', { Saw.ar(\freq.kr, 0.1) });
+Ndef('a').set('freq', 123);
+);
+"
+
+// anonymous nodeproxy - returns example code with globar var a = ...
```
Comment (muellmusik) [inline HelpSource/Tutorials/JITLib/jitlib_asCompileString.schelp]: global

Comment (muellmusik) [reply HelpSource/Tutorials/JITLib/jitlib_asCompileString.schelp]: Maybe should be interpreter var to be finicky.

Comment (telephon) [reply HelpSource/Tutorials/JITLib/jitlib_asCompileString.schelp]: fixed...

**Comment by telephon** [comment]
(comment: dismissing a review is really disencouraged by the interface, so there is an extra step in such light cases, sorry!)

**Pull request merged**